### PR TITLE
Support Crystal 0.34.0

### PR DESCRIPTION
**Pull request by marcosdsanchez**
### Tasks Done

- [X] Update the crystal version used by the docker image.
- [X] Errno was replaced by RuntimeError.
- [X] AtExitHandlers was moved to crystal namespace.
- [X] Logger class is deprecated.
---
### Todo

- [ ] Fix can't prove case is exhaustive warnings

.

### CODE REVIEW

**Comment by bew** [comment]
You'll also have to update the crystal version used by the docker image used for the build!

**Comment by marcosdsanchez** [comment]
> You'll also have to update the crystal version used by the docker image used for the build!
Currently its using 0.31.0

Done. Let me know it's fine. I'm not using Docker myself.

**Comment by marcosdsanchez** [comment]
Specs pass since https://github.com/crystal-lang-tools/scry/pull/170/commits/836950d4f5a54160b17813c049889e5625abb73e . The next commits are to fix warnings. I'm using this PR to learn Crystal so feedback is very welcome :smile:

**Comment by bew** [comment]
:tada: on making the tests go green!

About the new Log, see https://github.com/crystal-lang-tools/scry/pull/168, I've written a few comments there as well, you can try to apply them here if you want!

Otherwise we can merge the most important changes, and I'll try to apply my comments in another PR this week or weekend if I don't forget

File: src/scry/log.cr [hunk 1]
```diff
@@ -1,28 +1,8 @@
-require "logger"
+require "log"
 require "./client"
 
 module Scry
   module Log
-    class_property logger : Logger = Logger.new(nil)
-
-    class ClientLogger < Logger
```
Comment (bew) [inline src/scry/log.cr:8]: Warning! The new logger is not configured anymore, the log messages do not go anywhere..

You should have a custom backend that inherits from Log::IOBackend (I'm writing from memory), that writes messages to LSP with a "window/logMessage" event (see a few line below for example)

And then in the initialization code, this backend should be configured to receive all supported log messages.

Comment (marcosdsanchez) [reply src/scry/log.cr:8]: I initially tried inheriting from that class but run into issues. IIRC, the write method was changed to expect an `Entry` parameter. I'll try to make the changes that you suggest. Thank you very much for the feedback.

Comment (bew) [reply src/scry/log.cr:8]: Yes you now take an `Entry` structure that describes the log event, and you extract the fields relevant for you from it.

**Comment by marcosdsanchez** [comment]
Hey @bew I'm not sure I'll be able to work on this, this weekend. Feel free to cherry-pick the commits that you think are good and do the Log changes. I don't want to leave others waiting for me. I'm sorry about that. I'll try to work on this during the weekend if I find time.

**Comment by marcosdsanchez** [comment]
> Hey @bew I'm not sure I'll be able to work on this, this weekend. Feel free to cherry-pick the commits that you think are good and do the Log changes. I don't want to leave others waiting for me. I'm sorry about that. I'll try to work on this during the weekend if I find time.

I actually had some time :smile_cat:  . I made some changes. Let me know if that's what you planned.

**Comment by bew** [comment]
Thank you, it's exactly what I had in mind!

**Pull request merged**